### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by silviot**
This PR fixes the current failures seen on `master`.

The most interesting change is removing pinning from the bartlett1932 demo experiment.
Version pins are there to "freeze" the code that an experiment runs and ensure repeatability.
But the bartlett1932 demo experiment is used by dallinger in tests, and it always needs to run with the _current_ version to make sense.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -48,6 +48,12 @@
     'recommonmark',
 ]
 
+# Hack to make sphinx_js work with python 3.10
+import collections
+import collections.abc
+if not hasattr(collections, 'Mapping'):
+    collections.Mapping = collections.abc.Mapping
```
Comment (jessesnyder) [inline docs/source/conf.py:55]: My approach was to [fork sphinx-js](https://github.com/Dallinger/sphinx-js). The monkey-patch is definitely a lot less work, and I don't feel as uncomfortable with it as I thought I would, but I'm still undecided. Thoughts?

**Review by jessesnyder — COMMENTED** [review]
I'm still on the fence about how to deal with sphinx-js, but the rest of this looks to be (uncontroversially) a good idea.

File: demos/dlgr/demos/rogers/experiment.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ def submission_successful(self, participant):
         current_generation = participant.nodes()[0].generation
         if (
             num_approved % self.generation_size == 0
-            and (current_generation % 10 + 1) == 0
+            and (current_generation % 10 + 1) == 0  # noqa
```
[the remark this answers is not in the document]
Comment (silviot) [reply demos/dlgr/demos/rogers/experiment.py:128]: TBH I don't know exactly. All I know is that the same version of `flake8` gives different results on python 3.10 as opposed to python < 3.10.
I felt like sprinkling `# noqa`s around was good enough to get past this issue.
When looking up the problem I found [a SO answer](https://stackoverflow.com/a/57074416/41387) mentioning that the old recommended way is now considered an antipattern, and the new recommended way used to be considered an antipattern...

File: docs/source/conf.py (hunk 1 above)
Comment (silviot) [reply docs/source/conf.py:55]: If this change affected the dallinger codebase I'd be slightly more concerned, but it's only effective in the docs generation, so I believe it's safe.

**Review by jessesnyder — APPROVED** [review]

**Pull request merged**